consumer-test/consumerTestController.groovy

```diff
@@ -0,0 +1,213 @@
+import groovy.io.FileType
+
+import static groovy.json.JsonOutput.toJson
+
+COMMIT_HASH = null
+RUNNING_LOCALLY = false
+AUXILIARY_SLEEP_MS = 10000
+// Build is killed at 50 min, print log to console at minute 45
+PRINT_LOGS_AFTER_45_MINUTES_COUNTDOWN = (45 * 60 * 1000) / AUXILIARY_SLEEP_MS
+WORKSPACES_ROOT = 'workspaces'
+TEST_CASES_DIR = 'testCases'
+LIBRARY_VERSION_UNDER_TEST = null
+
+EXCLUDED_FROM_CONSUMER_TESTING_REGEXES = [
+    /^documentation\/.*$/,
+    /^.travis.yml$/,
+    /^test\/.*$/
+]
+
+
+if (!System.getenv('CX_INFRA_IT_CF_USERNAME') || !System.getenv('CX_INFRA_IT_CF_PASSWORD')) {
+    throw new RuntimeException('Environment variables CX_INFRA_IT_CF_USERNAME and CX_INFRA_IT_CF_PASSWORD need to be set.')
+}
+
+newEmptyDir(WORKSPACES_ROOT)
+TestRunnerThread.workspacesRootDir = WORKSPACES_ROOT
+LIBRARY_VERSION_UNDER_TEST = "git log --format=%H -n 1".execute().text.trim()
+TestRunnerThread.libraryVersionUnderTest = LIBRARY_VERSION_UNDER_TEST
+TestRunnerThread.repositoryUnderTest = System.getenv('TRAVIS_REPO_SLUG') ?: 'SAP/jenkins-library'
+
+def testCaseThreads
+def cli = new CliBuilder(
+    usage: 'groovy consumerTestController.groovy [<options>]',
+    header: 'Options:',
+    footer: 'If no options are set all tests are run centrally.')
+
+cli.with {
+    h longOpt: 'help', 'Print this help text and exit.'
+    l longOpt: 'run-locally', 'Run consumer tests locally.'
+    s(longOpt: 'single-test', args: 1, argName: 'filePath', 'Run single test.')
+}
+
+def options = cli.parse(args)
+
+if (options.h) {
+    cli.usage()
+    System.exit 0
+}
+
+if (options.l) {
+    RUNNING_LOCALLY = true
+}
+
+if (!RUNNING_LOCALLY) {
+    if (changeDoesNotNeedConsumerTesting()) {
+        notifyGithub("success", "No consumer tests necessary.")
+        println 'No consumer tests necessary.'
+        System.exit(0)
+    }
+
+    /*
+    In case the build is performed for a pull request TRAVIS_COMMIT is a merge
+    commit between the base branch and the PR branch HEAD. That commit is actually built.
+    But for notifying about a build status we need the commit which is currently
+    the HEAD of the PR branch.
+
+    In case the build is performed for a simple branch (not associated with a PR)
+    In this case there is no merge commit between any base branch and HEAD of a PR branch.
+    The commit which we need for notifying about a build status is in this case simply
+    TRAVIS_COMMIT itself.
+    */
+    COMMIT_HASH = System.getenv('TRAVIS_PULL_REQUEST_SHA') ?: System.getenv('TRAVIS_COMMIT')
+
+    notifyGithub("pending", "Consumer tests are in progress.")
+}
+
+if (options.s) {
+    testCaseThreads = [new TestRunnerThread(options.s)]
```

Comment:
What about a check: does the test case exist? Is the message we get in case the test case does not exist clear enough?